### DISCUSSION TITLE
What is Range Search?

### DISCUSSION THREAD

**Discussion by kanayer** [milvus-io/milvus#26611] [Q&A and General discussion]
Thank you for the release of the range search feature and its documentation! However, personally, I still find the explanation a little bit vague. Could you please answer the following questions:
1. When you set the `radius` to 0.8 and the `range_filter` value to 1.0, what do these values exactly mean? 
2. If we are considering IP distance and assume vectors are normalized (and therefore identical with cosine similarity), we know that vector values will be within bounds [-1, 1]. However, the `range_filter` definition in the documentation says that `range_filter` filters out most similar vectors with a distance greater than or equal to 1.0. How can the values be over 1.0? In other words, what are the value boundaries here?
3. Do I understand correctly, that `range_filter` is the top value filter and `radius` is the bottom value filter? For example, the vector values will be filtered between cos similarity values 0.8 and 1, correct? 
4.  Lastly, if I want to do a vector similarity search and I want to filter out vectors that have a cosine similarity score below a certain threshold (e.g. 0.5) but want to get the most similar vector values (no upper limit), what should I set `radius` and `range_filter` to?

**Comment by xiaofan-luan** [comment]
> Thank you for the release of the range search feature and its documentation! However, personally, I still find the explanation a little bit vague. Could you please answer the following questions:
> 
> 1. When you set the `radius` to 0.8 and the `range_filter` value to 1.0, what do these values exactly mean?
> 2. If we are considering IP distance and assume vectors are normalized (and therefore identical with cosine similarity), we know that vector values will be within bounds [-1, 1]. However, the `range_filter` definition in the documentation says that `range_filter` filters out most similar vectors with a distance greater than or equal to 1.0. How can the values be over 1.0? In other words, what are the value boundaries here?
> 3. Do I understand correctly, that `range_filter` is the top value filter and `radius` is the bottom value filter? For example, the vector values will be filtered between cos similarity values 0.8 and 1, correct?
> 4. Lastly, if I want to do a vector similarity search and I want to filter out vectors that have a cosine similarity score below a certain threshold (e.g. 0.5) but want to get the most similar vector values (no upper limit), what should I set `radius` and `range_filter` to?

1.different metrics has diffrent meanings, take IP as a example, this means you are looking for distance > 0.8, but smaller than 1.0. Usually you don't need to worry about range_filter, unless you want to exclude vectors very similar.

2. for IP the distance could be larger than 1.0. if you set range filter to 1.0 and used normalized vector, range filter won't has any effects.
